modules/tensor_mechanics/include/kernels/ADDynamicStressDivergenceTensors.h

```diff
@@ -0,0 +1,44 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADStressDivergenceTensors.h"
+
+// Forward Declarations
+template <ComputeStage>
+class ADDynamicStressDivergenceTensors;
+
+declareADValidParams(ADDynamicStressDivergenceTensors);
+
+/**
+ * ADDynamicStressDivergenceTensors is the automatic
+ * differentiation version of DynamicStressDivergenceTensors.
+ * This kernel derives from ADStressDivergenceTensors and
+ * adds stress related Rayleigh and HHT time integration terms.
+ */
+template <ComputeStage compute_stage>
+class ADDynamicStressDivergenceTensors : public ADStressDivergenceTensors<compute_stage>
+{
+public:
+  ADDynamicStressDivergenceTensors(const InputParameters & parameters);
+
+protected:
+  ADResidual computeQpResidual();
+
+  const MaterialProperty<RankTwoTensor> & _stress_older;
+  const MaterialProperty<RankTwoTensor> & _stress_old;
+
+  // Rayleigh damping parameter _zeta and HHT time integration parameter _alpha
+  const MaterialProperty<Real> & _zeta;
```

Comment:
If `_zeta` is expected to depend on non-linear variables this line would have to be 

```
const ADMaterialProperty(Real) & _zeta;
```

Reply:
Thanks for comment, @dschwen. _zeta is an input parameter that can be provided by the user as a function of space but it does not depend on the non-linear variables. So I have not made this change.